servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -487,6 +486,9 @@ private void terminateSubscriber(@Nullable Throwable cause) {
                     ChannelCloseUtils.close(channel, cause);
                 }
             }
+            // Notify listeners after the subscriber is terminated. Otherwise, WriteStreamSubscriber#channelClosed may
```

Comment:
If we invoke listeners before the write is complete, it will trigger `protocolPayloadEndOutboundSuccess` that initiates connection closure before the write is complete. As the result, the pipeline will see `ChannelOutputShutdownEvent` and trigger `WSS#channelClosed` changing the terminal signal from `onComplete` to `onError`.
`CloseHandler` does not know about the state of `WSS` to defer closure until the write is complete.

Reply:
Netty invokes listeners _after_ it processes the completion of the promise. I think we should do the same and first complete the subscriber, then notify listeners.